optimization2.jl

```diff
@@ -0,0 +1,539 @@
+        end
+
+        offsetidx += length(ref_arcs_to)
+
+        @assert offsetidx == length(ret)
+        return
+    end
+
+    con_lbs = Float64[]
+    con_ubs = Float64[]
+
+    #@constraint(model, va[i] == 0)
+    for (i,bus) in ref[:ref_buses]
+        push!(con_lbs, 0.0)
+        push!(con_ubs, 0.0)
+    end
+
+    #power_balance_p_con
+    for (i,bus) in ref[:bus]
+        push!(con_lbs, 0.0)
+        push!(con_ubs, 0.0)
+    end
+
+    #power_balance_q_con
+    for (i,bus) in ref[:bus]
+        push!(con_lbs, 0.0)
+        push!(con_ubs, 0.0)
+    end
+
+    #power_flow_p_from_con
+    for (l,i,j) in ref[:arcs_from]
+        push!(con_lbs, 0.0)
+        push!(con_ubs, 0.0)
+    end
+
+    #power_flow_p_to_con
+    for (l,i,j) in ref[:arcs_to]
+        push!(con_lbs, 0.0)
+        push!(con_ubs, 0.0)
+    end
+
+    #power_flow_q_from_con
+    for (l,i,j) in ref[:arcs_from]
+        push!(con_lbs, 0.0)
+        push!(con_ubs, 0.0)
+    end
+
+    #power_flow_q_to_con
+    for (l,i,j) in ref[:arcs_to]
+        push!(con_lbs, 0.0)
+        push!(con_ubs, 0.0)
+    end
+
+    #power_flow_vad_con
+    for (l,i,j) in ref[:arcs_from]
+        branch = ref[:branch][l]
+        push!(con_lbs, branch["angmin"])
+        push!(con_ubs, branch["angmax"])
+    end
+
+    #power_flow_mva_from_con
+    for (l,i,j) in ref[:arcs_from]
+        branch = ref[:branch][l]
+        push!(con_lbs, -Inf)
+        push!(con_ubs, branch["rate_a"]^2)
+    end
+
+    #power_flow_mva_to_con
+    for (l,i,j) in ref[:arcs_to]
+        branch = ref[:branch][l]
+        push!(con_lbs, -Inf)
+        push!(con_ubs, branch["rate_a"]^2)
+    end
+
+    model_variables = length(var_init)
+    ret = Array{Float64}(undef, length(con_lbs))
+    model_constraints = length(con_lbs)
+
+    optf = Optimization.OptimizationFunction(opf_objective, adchoice; cons=opf_constraints)
+    optprob = Optimization.instantiate_function(optf, var_init, adchoice, nothing, length(con_lbs))
+    optprob = Optimization.OptimizationFunction(opf_objective, adchoice; cons=opf_constraints,
+    grad = optprob.grad, hess = optprob.hess, cons_j = optprob.cons_j, cons_h = optprob.cons_h,
+    hess_prototype = optprob.hess_prototype, cons_jac_prototype = optprob.cons_jac_prototype, cons_hess_prototype = optprob.cons_hess_prototype,
+    hess_colorvec = optprob.hess_colorvec, cons_jac_colorvec = optprob.cons_jac_colorvec, cons_hess_colorvec = optprob.cons_hess_colorvec)
+    return Optimization.OptimizationProblem(
+        optprob,
+        var_init;
+        lb=var_lb,
+        ub=var_ub,
+        lcons=con_lbs,
+        ucons=con_ubs,
+    )
+end
+
+function solve_opf(file_name)
+    start = time()
+    dataset = load_and_setup_data(file_name);
+    data_load_time = time() - start
+    prob = build_opf_optimization_prob(dataset)
+    model_build_time = time() - start - data_load_time
+    sol = Optimization.solve(prob, Ipopt.Optimizer())
```

Comment:
```suggestion
    sol = Optimization.solve(prob, OptimizationMOI.MOI.OptimizerWithAttributes(Ipopt.Optimizer, "linear_solver" => "ma27"))
```

@ccoffrin this should work to set the linear solver

Reply:
Thanks! In retrospect, should have been totally obvious to me this is how it would work.